clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -184,8 +184,8 @@ public String toString() {
 
     public static final class PartitionData {
         public final long timestamp;
-        @Deprecated
-        public final int maxNumOffsets; // only supported in v0
+        // Note, this maximum number offsets is only used in ListOffsetRequest v0
+        public final int maxNumOffsets;
```

Comment:
Removed the `@Deprecated` annotation since both the Kafka broker and the Kafka client need to support old version of the RPC.

Reply:
Nit: `Note, this maximum number offsets` is redundant right? Seems like "Only used in ListOffsetRequest v0" would be a more concise version and achieve the same thing.